### ISSUE TITLE
Queued Requests should be flushed when a dapp connects or disconnects.

### ISSUE THREAD

**Issue by BelfordZ** [MetaMask/core#4152]
See this comment for more details / context: https://github.com/MetaMask/core/pull/4139#discussion_r1561157147

**Comment by BelfordZ** [comment]
update: It shouldnt matter in relation to connecting, only disconnecting.

**Comment by BelfordZ** [comment]
another update: there is currently a bug where if you call revokePermissions while there is a pending confirmation, the confirmation is not cleared. 

This applies when either queueing is enabled or disabled.

**Comment by BelfordZ** [comment]
Todo:

- if a dapp disconnects, but theres requests from several dapps, when the queueing is turned off it will clear all requests (not just the ones for the currently focused dapp)
- with queuing on, it will clear only the requests for the dapp who was disconnected

**Comment by BelfordZ** [comment]
Questions:

 - how should we trigger the flushing / clearing of pending confirms? Should we listen to changes on permissionController, or directly call a method/helper from inside the revokePermissions handler? -> the stateChange event should do the trick, but I really think we should be cautious of using patches for this one.

**Comment by adonesky1** [comment]
> When a dapp disconnects, aka the permissions are removed for the dapp, are we currently unsetting / deleting the value stored for the domain in the selectedNetworkController?

yes we are here: https://github.com/MetaMask/core/blob/95d02fae36ab4229069e6dccf0dd9c27b8e60a99/packages/selected-network-controller/src/SelectedNetworkController.ts#L176
